ipapython/ipaldap.py

```diff
@@ -1245,11 +1245,10 @@ def make_filter_from_attr(
             return cls.combine_filters(flts, rules)
         elif value is not None:
             if isinstance(value, bytes):
-                if six.PY3:
-                    value = value.decode('raw_unicode_escape')
+                value = u''.join(
+                    u'\\{:02x}'.format(i) for i in six.iterbytes(value))
```

Comment:
format() with hex is rather inefficient and slow. An implementation with hexlify() is five to ten times faster.

```
def enc_hexlify(s):
    hexstring = binascii.hexlify(s).decode('ascii')
    # [-2:0] is empty string for the initial \
    return u'\\'.join(hexstring[i:i+2] for i in six.moves.range(-2, len(s)+2, 2))
```